jenkins/automation/rgw/ceph/cortx-rgw-build-container-image.groovy

```diff
@@ -0,0 +1,137 @@
+pipeline {
+    agent {
+        node {
+            // label 'docker-image-builder-centos-7.9.2009'
+            label 'build-retention'
+        }
+    }
+
+    options {
+        timeout(time: 240, unit: 'MINUTES')
+        timestamps()
+        buildDiscarder(logRotator(daysToKeepStr: '30', numToKeepStr: '30'))
+        ansiColor('xterm')
+    }
+
+    environment {
+        ARCH = "x86_64"
+        // CEPH_PROJECT = "ceph"
+        // REGISTRY = "cortx-docker.colo.seagate.com"
+        // LOCAL_REG_CRED = credentials('local-registry-access')
+        CEPH_PROJECT = "cortx-rgw"
+        REGISTRY = "ssc-vm-g4-rhev4-1774.colo.seagate.com"
+        LOCAL_REG_CRED = credentials('dev-harbor')
+    }
+
+    parameters {
+        string(name: 'CEPH_CONTAINER_REPO', defaultValue: 'https://github.com/nitisdev/ceph-container/', description: 'Repository for ceph container image scripts.', trim: true)
+        string(name: 'CEPH_CONTAINER_BRANCH', defaultValue: 'rgw-rockylinux-test', description: 'Branch or GitHash for ceph container image scripts.', trim: true)
+        string(name: 'CEPH_RELEASE', defaultValue: 'quincy', description: 'Ceph release to build image from.', trim: true)
+        string(name: 'OS_IMAGE', defaultValue: 'rockylinux', description: 'Base OS docker image to build from.', trim: true)
+        string(name: 'OS_IMAGE_TAG', defaultValue: '8', description: 'OS docker image tag.', trim: true)
+    }    
+
+    stages {
+        stage ('Checkout ceph-container repo') {
+            steps { 
+                cleanWs()            
+                script {
+                    checkout([$class: 'GitSCM', branches: [[name: "${CEPH_CONTAINER_BRANCH}"]], doGenerateSubmoduleConfigurations: false, extensions: [], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'cortx-admin-github', url: "${CEPH_CONTAINER_REPO}"]]])                
+                }
+            }
+        }
+
+        stage ('Build Ceph Container Image') {
+            steps {
+                script { build_stage = env.STAGE_NAME }
+                sh label: 'Build Ceph Container Image', script: """
+                    echo "Starting Build"
+                    make FLAVORS=${CEPH_RELEASE},${OS_IMAGE},${OS_IMAGE_TAG} RELEASE=${CEPH_CONTAINER_BRANCH} build
+
+                    echo -e "==============================\n"
+
+                    echo "List created images:"
+                    # docker images --format "{{.Repository}}:{{.Tag}}" --filter reference=${CEPH_PROJECT}/daemon-base:${CEPH_CONTAINER_BRANCH}-${CEPH_RELEASE}-${OS_IMAGE}-${OS_IMAGE_TAG}-${ARCH}
```

Comment:
remove commented code if not required

Reply:
This is a test job for future reference. Currently it will be disabled. All comments are placeholders for funtionality.